bootstrap/src/basis/map.ml

```diff
@@ -1552,6 +1566,31 @@ let%expect_test "of_alist,to_alist" =
     of_alist [(0, "0"); (1, "1"); (66, "66"); (91, "91")]; to_alist -> [(1, "1"); (91, "91"); (66, "66"); (0, "0")]
     |}]
 
+let%expect_test "choose_hlt" =
+  let open Format in
+  printf "@[";
+  (* test is n^2 time complexity, so keep n small. *)
+  let rec test n i map = begin
+    match i < n with
+    | false -> map
+    | true -> begin
+        validate map;
+        let map' = test n (succ i) (insert_hlt ~k:i ~v:(i * 100) map) in
+        let k, v = choose_hlt map' in
+        assert (k * 100 = v);
+        let map'' = remove_hlt k map' in
+        validate map'';
+        assert ((length map') = (length map'') + 1);
+        map''
+      end
+  end in
+  let e = empty (module UsizeTestCmper) in
+  let _ = test 100 0 e in
+  printf "@]";
+
+  [%expect{|
+    |}]
```

Comment:
Guessing it doesn't make sense to print and check for the result of `choose_hlt` since the result of `choose` changes based on the hash seed? Wondering what we can do here to make it glaringly obvious the test really did something.